torchinductor/lowering.py

```diff
@@ -2877,6 +2877,20 @@ def fn(*args):
     )
 
 
+# TODO(lezcano) I believe the casting behaviour of prims.div is wrong
+# https://github.com/pytorch/pytorch/issues/84412
+# div prim performs truncation division on integer inputs
+#   and true division for floating and complex inputs
+@register_lowering([prims.div], broadcast=True)
+def div_prim(a, b):
+    is_integral = isinstance(a, (bool, int)) or is_integer_type(a)
```

Comment:
is_integer_type already checks int. And we have an is_boolean_type

maybe
```suggestion
    is_integral = is_boolean_type(a) or is_integer_type(a)
```

If you want to be broader (numpy types, custom integral types), consider making an is_integral_type that looks like is_integer_type but calls into `isinstance(a, numbers.Integral)`. No strong opinion.

Reply:
I copied this from primtorch and I didn't think too much about it. Fixed!

Reply:
LGTM - let @jansel take a look too.